### PR TITLE
feat: deprecate CL:0000003 'native cell' and replace with "unknown"

### PR DESCRIPTION
**Pull request by danieljhegeman**
- add exception for "unknown" to schema def yml
- update test_convert_template.py
- migrate script should get auto-updated during migration workflow

## Reason for Change

- #735 

## Changes

- Permit `"unknown"` for `cell_type_ontology_term_id`
  - Annotate `cell_type` as `"unknown"`
- Permit `"unknown"` for `tissue_ontology_term_id` when `tissue_type` is `"cell culture"`
  - Annotate `tissue` as `"unknown"`
- Various unrelated typo fixes
- Fix test `unmigrated_X` in `examples_validate.py` to reference correct obs df shape as source of row count when instantiating

## Testing

- unit tests added to test against scenarios where the ontology term for cell type or tissue is `"unknown"`
- test label writes
```
pip install git+https://github.com/chanzuckerberg/single-cell-curation/@main#subdirectory=cellxgene_schema_cli
```

## Notes for Reviewer

### CODE REVIEW

File: cellxgene_schema_cli/cellxgene_schema/validate.py [hunk 1]
```diff
@@ -1255,9 +1255,9 @@ def _check_column_availability(self):
 
             # Do it for columns that map to other columns, for post-upload annotation
             if "columns" in component_def:
-                for _column, columns_def in component_def["columns"].items():
-                    if "add_labels" in columns_def:
-                        self._check_single_column_availability(component, columns_def["add_labels"])
+                for column_def in component_def["columns"].values():
+                    if "add_labels" in column_def:
+                        self._check_single_column_availability(component, column_def["add_labels"])
```
Comment (danieljhegeman) [inline cellxgene_schema_cli/cellxgene_schema/validate.py:1260]: unrelated code refactor

File: cellxgene_schema_cli/tests/fixtures/examples_validate.py [hunk 2]
```diff
@@ -297,12 +297,12 @@
     ],
 )
 
-unmigrated_X = numpy.zeros([good_obs.shape[0], var_unmigrated.shape[0]], dtype=numpy.float32)
-adata_with_lables_unmigrated = anndata.AnnData(
+unmigrated_X = numpy.zeros([unmigrated_obs.shape[0], var_unmigrated.shape[0]], dtype=numpy.float32)
```
Comment (danieljhegeman) [inline cellxgene_schema_cli/tests/fixtures/examples_validate.py:300]: I believe this was what you intended @Bento007 (not the typo fix from `umigrated_obs` => `unmigrated_obs`, but the reference to the unmigrated obs df rather than the "good" obs df). But everything was working by chance since each df had 2 rows.

Confirm?

File: cellxgene_schema_cli/tests/test_schema_compliance.py [hunk 3]
```diff
@@ -576,6 +576,48 @@ def test_self_reported_ethnicity_ontology_term_id__unknown(self, validator_with_
...
 
+    def test_cell_type_ontology_term_id__unknown(self, validator_with_adata):
+        """
+        Test 'unknown' cell_type_ontology_term_id is valid
+        """
+        validator = validator_with_adata
+        obs = validator.adata.obs
+        obs.at["Y", "cell_type_ontology_term_id"] = "unknown"
+        assert validator.validate_adata()
+        assert validator.errors == []
```
Comment (joyceyan) [inline cellxgene_schema_cli/tests/test_schema_compliance.py:587]: i think this line is redundant since `validator.validate_adata()` should only return True if there's no errors

File: cellxgene_schema_cli/tests/test_schema_compliance.py [hunk 4]
```diff
@@ -576,6 +576,48 @@ def test_self_reported_ethnicity_ontology_term_id__unknown(self, validator_with_
...
+        # Arrange -- relies on "tissue_type" value for index "Y" being "cell culture", set explicitly
+        obs.at["Y", "tissue_type"] = "cell culture"
+        obs.at["Y", "tissue_ontology_term_id"] = "unknown"
+
+        assert validator.validate_adata()
+        assert validator.errors == []
+
+    def test_tissue_ontology_term_id__unknown_invalid(self, validator_with_adata):
+        """
+        Test 'unknown' tissue_ontology_term_id is valid if tissue_type is 'cell culture'
```
Comment (joyceyan) [inline cellxgene_schema_cli/tests/test_schema_compliance.py:605]: i'm guessing this comment was copied and pasted, but can you update this to reflect what the test actually does? it would also be fine to remove this entirely, since the comment on line 610 explains the test sufficiently imo

**Review by joyceyan — CHANGES_REQUESTED** [review]

File: cellxgene_schema_cli/tests/test_schema_compliance.py (hunk 4 above)
Comment (danieljhegeman) [reply cellxgene_schema_cli/tests/test_schema_compliance.py:605]: thank you 🙏 😅

File: cellxgene_schema_cli/tests/test_schema_compliance.py (hunk 3 above)
Comment (danieljhegeman) [reply cellxgene_schema_cli/tests/test_schema_compliance.py:587]: Fair, but prefer to leave it explicit esp since this is the pattern used by other tests

**Review by joyceyan — APPROVED** [review]
